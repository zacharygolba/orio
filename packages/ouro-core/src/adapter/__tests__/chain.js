// @flow

import Chain from '../chain'
import * as ouro from '../../'

let subj

beforeEach(() => {
  const producerA = ouro.of(1, 2, 3).producer
  const producerB = ouro.of(4, 5, 6).producer
  subj = new Chain(producerA, producerB)
})

test('#@@iterator()', () => {
  for (const item of subj) {
    expect(item).toMatchSnapshot()
  }
})

test('#next()', () => {
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
})
