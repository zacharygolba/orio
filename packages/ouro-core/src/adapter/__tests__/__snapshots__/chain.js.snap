// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#@@iterator() 1`] = `1`;

exports[`#@@iterator() 2`] = `2`;

exports[`#@@iterator() 3`] = `3`;

exports[`#@@iterator() 4`] = `4`;

exports[`#@@iterator() 5`] = `5`;

exports[`#@@iterator() 6`] = `6`;

exports[`#next() 1`] = `
Object {
  "done": false,
  "value": 1,
}
`;

exports[`#next() 2`] = `
Object {
  "done": false,
  "value": 2,
}
`;

exports[`#next() 3`] = `
Object {
  "done": false,
  "value": 3,
}
`;

exports[`#next() 4`] = `
Object {
  "done": false,
  "value": 4,
}
`;

exports[`#next() 5`] = `
Object {
  "done": false,
  "value": 5,
}
`;

exports[`#next() 6`] = `
Object {
  "done": false,
  "value": 6,
}
`;

exports[`#next() 7`] = `
Object {
  "done": true,
  "value": undefined,
}
`;
